src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallRecords/RegistryWorkloadInstallationRecordRepository.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.DotNet.Installer.Windows;
+using Microsoft.Win32;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install.InstallRecord
+{
+    /// <summary>
+    /// Provides support for reading and writing workload installation records in the registry
+    /// for MSI based workloads.
+    /// </summary>
+#if NETCOREAPP
+    [SupportedOSPlatform("windows")]
+#endif
+    internal class RegistryWorkloadInstallationRecordRepository : InstallerBase, IWorkloadInstallationRecordRepository
+    {
+        /// <summary>
+        /// The base path of workload installation records in the registry.
+        /// </summary>
+        internal readonly string BasePath = @"SOFTWARE\Microsoft\dotnet\InstalledWorkloads\Standalone";
+
+        /// <summary>
+        /// The base key to use when reading/writing records.
+        /// </summary>
+        private RegistryKey _baseKey = Registry.LocalMachine;
+
+
+        internal RegistryWorkloadInstallationRecordRepository(InstallElevationContextBase elevationContext, ISetupLogger logger)
+            : base(elevationContext, logger)
+        {
+
+        }
+
+        /// <summary>
+        /// Constructor for testing purposes to allow changing the base key from HKLM.
+        /// </summary>
+        /// <param name="baseKey">The base key to use, e.g. <see cref="Registry.CurrentUser"/>.</param>
+        internal RegistryWorkloadInstallationRecordRepository(InstallElevationContextBase elevationContext, ISetupLogger logger,
+            RegistryKey baseKey, string basePath) 
+            : this(elevationContext, logger)
+        {
+            _baseKey = baseKey;
+            BasePath = basePath;
+        }
+
+        public void DeleteWorkloadInstallationRecord(WorkloadId workloadId, SdkFeatureBand sdkFeatureBand)
+        {
+            Elevate();
+
+            if (IsElevated)
+            {
+                string workloadInstallationKeyName = Path.Combine(BasePath, $"{sdkFeatureBand}", $"{workloadId}");
+                Log?.LogMessage($"Deleting {workloadInstallationKeyName}");
+                _baseKey.DeleteSubKeyTree(workloadInstallationKeyName, throwOnMissingSubKey: false);
+            }
+            else if (IsClient)
+            {
+                Dispatcher.SendWorkloadRecordRequest(InstallRequestType.DeleteWorkloadInstallationRecord,
+                    workloadId, sdkFeatureBand);
+
+                //LogAndExitOnFailure(response, $"Failed to delete workload record key.");
```

Comment:
nit: remove commented code

Reply:
Actually that's a bug, good catch. I changed some of the error reporting a while back and I think I missed this class. I'll update the PR.